worker/caasapplicationprovisioner/ops.go

```diff
@@ -697,7 +697,9 @@ func ensureScale(appName string, app caas.Application, appLife life.Value,
 	if ps.ScaleTarget >= len(units) {
 		logger.Infof("scaling application %q to desired scale %d", appName, ps.ScaleTarget)
 		err = app.Scale(ps.ScaleTarget)
-		if err != nil {
+		if appLife != life.Alive && errors.Is(err, errors.NotFound) {
+			logger.Infof("dying application %q is already removed", appName)
```

Comment:
So I think we should only do this if ScaleTarget is actually 0 and still error otherwise.

Reply:
I had considered that, but since an app is already dying/dead and the units+pods have gone. This is safe.